### PR TITLE
feat(observer): complete runtime, sync, API, and Web4 contracts

### LINKED ISSUE #1973: Spec: Standalone Observer Node Startup, Sync, Lifecycle, and Service Operation
## Background

Issue #1878 established that observer nodes currently exist mostly as a `FullNode` startup path with `NodeRole::Observer`.

Today an observer can partially operate as a full non-validator node, but this is not yet a first-class standalone product path.
The current implementation still has several gaps:
- observer mode is mostly implicit rather than explicitly productized
- startup/join behavior depends heavily on bootstrap peer assumptions
- sync logic is observer-specific but still narrow and bootstrap-peer-centric
- several runtime paths still assume validators will eventually mine/broadcast the effects
- observer API endpoints are stubbed
- observer lifecycle / health / operator visibility is incomplete
- there is no dedicated end-to-end observer operational test matrix

This issue is explicitly **not** about observer rewards. Rewards are tracked under #1964.
This issue is about making an observer node able to spawn, connect, sync, restart, and operate safely on its own.

## Goal

Make observer mode a first-class standalone node operation mode that can:
- start from clean state
- connect to a network
- bootstrap and catch up safely
- maintain canonical full local state
- restart safely
- expose operational health/status
- serve observer-appropriate APIs and content services
- run indefinitely without validator-only assumptions

## Scope

Covers:
- observer startup contract
- observer join/bootstrap policy
- observer sync and catch-up behavior
- observer lifecycle state model
- observer-safe runtime behavior
- observer operational APIs and health surfaces
- optional content/Web4-serving integration as an observer capability
- end-to-end observer tests and operator documentation

Does not cover:
- observer reward economics
- canonical reward accounting
- staking/reward payout design

## Design Principles

1. Observer is a real operational mode
- not just a side effect of disabling validators
- startup, logs, health, APIs, and tests must treat it explicitly

2. Observer must never accidentally act like a validator
- no consensus engine
- no block production
- no validator-only mutation paths

3. Observer must not accidentally create genesis when it should join an existing network
- join/bootstrap policy must be explicit and deterministic

4. Observer must survive restart and catch-up safely
- recovery behavior is part of the definition of done

5. Observer services should be layered
- base observer: full-chain sync + verification + operational APIs
- optional storage observer: stores content
- optional gateway observer: serves Web4/public content

### 1. First-class observer startup contract
Clarify and enforce the startup contract for observer nodes:
- required config
- no validator keys required
- full blockchain required
- no consensus startup
- no local block production

### 2. Observer bootstrap/join policy
Define exact behavior for:
- empty disk + existing network
- empty disk + no network
- local chain + reachable peers
- local chain + no peers
- stale local chain + newer peers

Observer should not silently create genesis in cases where it should wait or join.

### 3. Observer sync hardening
Strengthen:
- peer source selection
- retry/failover behavior
- catch-up progress visibility
- divergence handling
- restart resumption
- peer unavailability handling

### 4. Runtime observer safety audit
Audit and fix runtime flows that still rely on validator-side mining/broadcast assumptions.
Examples include startup registration/backfill paths and any deferred state assumptions.

### 5. Observer operational API surface
Add real observer operational endpoints for:
- startup phase / current lifecycle state
- sync status
- local height vs peer height
- current chain source / peer source
- lag / degraded mode indicators
- content-serving capability state where applicable

This is separate from the more advanced consensus-observer analytics tracked elsewhere.

### 6. Observer lifecycle state model
Define explicit operational states such as:
- Starting
- Discovering
- Bootstrapping
- Syncing
- CaughtUp
- Degraded
- Serving
- Stopped

### 7. Content/Web4 observer capability definition
Explicitly define whether observer nodes may:
- store content
- serve Web4/public content
- act as HTTPS/Web4 gateways
- serve DHT-backed content

If yes, separate that from the minimal observer contract so the mode is composable.

### 8. Test and operator-readiness gates
Add the full observer test matrix and operator docs.

## Acceptance Criteria

1. A fresh observer node can start from clean state and join an existing network safely.
2. Observer mode never starts consensus engine or block production.
3. Observer catches up after lag or restart without validator-only assumptions.
4. Observer has explicit operational health/status surfaces.
5. Observer content-serving behavior is either explicitly supported and tested or explicitly excluded.
6. End-to-end tests cover startup, join, restart, lag recovery, and safe operation.
7. Operator documentation exists for running observer nodes in practice.

## Suggested Sub-Issues

- observer startup contract and config hardening
- observer bootstrap/join policy
- observer sync/failover hardening
- runtime validator-assumption cleanup for observer mode
- observer operational APIs and lifecycle states
- observer content/Web4 capability definition and integration
- observer E2E tests and docs

## References
- Issue #1878
- `zhtp/src/runtime/mod.rs`
- `zhtp/src/runtime/components/blockchain.rs`
- `zhtp/src/runtime/components/consensus.rs`
- `zhtp/src/runtime/components/identity.rs`
- `zhtp/src/runtime/node_runtime.rs`

### LINKED ISSUE #1974: Observer Runtime: Harden startup contract and explicit observer config path
Parent: #1973

## Goal
Make observer startup an explicit, supported operational path rather than an implicit side effect of disabling validators.

## Scope
Clarify and enforce:
- required observer config
- no validator key dependency
- no consensus startup
- no local block production
- full-chain observer expectations
- log and error messaging that clearly identifies observer mode

## Acceptance Criteria
- Observer startup contract is explicit in code and config handling.
- Observer mode does not require validator-only material.
- Startup paths and logs consistently identify observer role.
- Tests cover correct observer-mode initialization.

### LINKED ISSUE #1975: Observer Runtime: Define bootstrap/join policy and prevent accidental genesis
Parent: #1973
Depends on: #1974

## Goal
Define deterministic observer behavior for joining an existing network versus waiting versus local-only startup.

## Scope
Handle explicitly:
- empty disk + existing network
- empty disk + no network
- local chain + peers
- stale local chain + newer peers
- bootstrap leader edge cases as they affect observer behavior

## Acceptance Criteria
- Observer does not accidentally create genesis when it should join or wait.
- Join/bootstrap rules are explicit and deterministic.
- Tests cover empty-state join, no-peer behavior, and stale-chain recovery decisions.

### LINKED ISSUE #1976: Observer Runtime: Harden sync, failover, catch-up, and restart recovery
Parent: #1973
Depends on: #1975

## Goal
Make observer sync robust enough for real standalone operation.

## Scope
Strengthen:
- peer source selection
- bootstrap-peer failover
- catch-up progress visibility
- divergence / ahead-peer handling
- restart resumption
- lag recovery after network partition or downtime

## Acceptance Criteria
- Observer can start empty and catch up reliably.
- Observer can fall behind and recover.
- Observer can fail over between peers when primary bootstrap sources fail.
- Tests cover restart recovery, lag recovery, and peer failover.

### LINKED ISSUE #1977: Observer Runtime: Remove validator-only assumptions from observer startup and background flows
Parent: #1973
Depends on: #1974, #1975

## Goal
Audit and fix runtime paths that still assume validator-side mining or broadcast behavior when running as observer.

## Scope
Review observer safety in flows such as:
- wallet registration bootstrap
- SOV backfill / startup tx handling
- deferred pending-state assumptions
- any other path that assumes validator-produced blocks will complete local intent

## Acceptance Criteria
- Observer mode no longer relies on hidden validator-only assumptions to remain healthy.
- Observer-safe alternatives are explicit.
- Tests cover observer startup/background flows that previously depended on validator behavior.

### LINKED ISSUE #1978: Observer Runtime: Add operational APIs and explicit lifecycle states
Parent: #1973
Depends on: #1976, #1977

## Goal
Give observer nodes real operational visibility and a first-class lifecycle model.

## Scope
Implement:
- explicit lifecycle states such as Starting, Discovering, Bootstrapping, Syncing, CaughtUp, Degraded, Serving, Stopped
- operational observer endpoints for sync status, lag, chain source, peer source, and degraded state
- health/status reporting suitable for operators

Note: this is separate from advanced consensus-observer analytics.

## Acceptance Criteria
- Observer lifecycle state model exists and is surfaced.
- Operational observer endpoints return real data rather than placeholders.
- Operators can determine whether an observer is starting, syncing, caught up, or degraded.

### LINKED ISSUE #1979: Observer Runtime: Define and integrate content/Web4 serving capability for observer nodes
Parent: #1973
Depends on: #1976, #1978

## Goal
Explicitly define whether and how observer nodes store and serve content, including Web4/public gateway behavior.

## Scope
Clarify and implement:
- minimal observer contract versus storage observer / gateway observer capabilities
- whether observer nodes may store content replicas
- whether observer nodes may serve Web4/public content
- whether observer nodes may act as HTTPS/Web4 gateways
- DHT-backed content serving expectations

## Acceptance Criteria
- Observer content-serving capability is explicit, not accidental.
- The distinction between base observer and content-serving observer roles is documented in code and behavior.
- Tests cover supported content/Web4 observer paths, or the unsupported paths are explicitly rejected.

### LINKED ISSUE #1980: Observer Runtime: Add end-to-end startup, sync, restart, and operator-readiness tests/docs
Parent: #1973
Depends on: #1974, #1975, #1976, #1977, #1978, #1979

## Goal
Add the observer-mode readiness gates required to treat observer nodes as a real supported product path.

## Scope
Add tests and docs for:
- clean observer boot into an existing network
- observer restart from persisted state
- observer lag and catch-up recovery
- observer failover when bootstrap peers are unavailable
- observer never starting consensus engine
- observer never producing blocks
- observer service/API health during sync and steady state
- content/Web4 serving behavior if supported
- operator runbook for configuring and running observer nodes

## Acceptance Criteria
- End-to-end observer tests exist and pass.
- Operator documentation exists for real deployment.
- Release/readiness gates explicitly include observer operation.

### PR DESCRIPTION
**Pull request by umwelt**
## Summary
- enforce the explicit full-node observer startup contract and join-only startup behavior
- block accidental genesis for empty join-only observers and keep discovery active until a network is found or local chain state exists
- skip validator-only identity backfill on non-validator startup/background paths
- refresh observer sync peer candidates each round and continue past stale, failed, or only-partially-helpful peers
- keep observer sync failover active across catch-up rounds and restart visibility
- expose real observer operational APIs for status, sync, network health, and lifecycle state
- document observer deployment and readiness checks in an operator runbook
- define observer Web4 layering explicitly and route content/gateway reads through the canonical content service
- add observer startup, sync, restart, Web4, and multi-orchestrator regression coverage across the in-repo harness

## Delivered Scope
- closes #1974: startup contract and explicit observer config/runtime path
- closes #1975: deterministic bootstrap/join policy and accidental-genesis prevention
- closes #1976: sync, failover, catch-up, and restart recovery hardening
- closes #1977: validator-only startup/background assumption cleanup
- closes #1978: operational APIs and explicit lifecycle state exposure
- closes #1979: observer content/Web4 and gateway capability definition/integration
- closes #1980: observer readiness tests and operator documentation within the current in-repo harness
- closes #1973: standalone observer node startup, sync, lifecycle, and service operation spec delivery

## Remaining Non-Blocking Follow-Up
- no external multi-process peer-mesh harness was added in this PR
- current readiness coverage reaches orchestrator/runtime and handler/gateway level, including multi-orchestrator shared-state flows, which is the practical ceiling of the existing in-repo harness
- a future dedicated process/network harness can be added separately if we want CI coverage over real sockets and isolated node processes

## Verification
- cargo test -p zhtp runtime::components::blockchain::tests:: -- --nocapture
- cargo test -p zhtp runtime_orchestrator_tests -- --nocapture
- cargo test -p zhtp observer_network_health -- --nocapture
- cargo test -p zhtp observer_lifecycle -- --nocapture
- cargo test -p zhtp web4_handler -- --nocapture
- cargo test -p zhtp gateway_serves_registered_observer_content_from_host_header -- --nocapture
- cargo test -p zhtp observer_multi_orchestrator_join_sync_restart_sequence_preserves_shared_network_state -- --nocapture
- cargo test -p zhtp observer -- --nocapture

### CODE REVIEW

File: zhtp/src/runtime/mod.rs [hunk 1]
```diff
@@ -1210,6 +1210,53 @@ impl RuntimeOrchestrator {
...
+            .unwrap_or_default();
+
+        Err(anyhow::anyhow!(
+            "Observer/full-node startup requires an existing network with committed blocks. \
+             Local chain state is empty, discovered peers={}, discovered_height={}, \
+             configured_bootstrap_peers={}. Refusing to create genesis in observer mode.",
+            discovered_peer_count,
+            discovered_height,
+            configured_bootstrap_peers
+        ))
```
[the remark this answers is not in the document]
Comment (umwelt) [reply zhtp/src/runtime/mod.rs:1250]: Addressed in 47d4d62d. The join-policy behavior remains intentionally strict, but the wording and regression coverage now make that contract explicit: a genesis-only peer set is not considered ready for a fresh observer join, and the error now says "committed blocks beyond genesis".
